File: project-euler/ruby/12.rb
#!/usr/bin/env ruby

def ndivs(n)
	return 1 if(n < 2) 
	p = 2
	c = 1
	while p**2 < n
		c += 2 if n % p == 0
		p += 1
	end
	c += 1 if p**2 == n
	return c
end

n = 1
t = n

until ndivs(n) > 500 
	t += 1
	n += t
end

puts n
# 76576500

File: project-euler/python/12.py
#!/usr/bin/env python

from math import sqrt

def gentris():
	i, n = 1, 1
	while True:
		yield n
		i += 1
		n += i

def ndivs(n):
	if n < 2: return 1
	d = 2
	sqt = int(sqrt(n))
	if sqt * sqt == n:
		c = 3
	else:
		c = 2
	for i in range(2,sqt):
		if n % i == 0:
			c += 2
	return c

def genlen(gen):
	l = 0
	for i in gen:
		l += 1
	return l

if name == __main__:
	en = gentris()


